plugins/vppv2/aclplugin/descriptor/acl.go

```diff
@@ -0,0 +1,397 @@
+//  Copyright (c) 2018 Cisco and/or its affiliates.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at:
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package descriptor
+
+import (
+	"strings"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/ligato/cn-infra/idxmap"
+	"github.com/ligato/cn-infra/logging"
+	"github.com/ligato/cn-infra/logging/logrus"
+	"github.com/ligato/vpp-agent/plugins/kvscheduler/api"
+	"github.com/ligato/vpp-agent/plugins/vppv2/aclplugin/aclidx"
+	"github.com/ligato/vpp-agent/plugins/vppv2/aclplugin/descriptor/adapter"
+	"github.com/ligato/vpp-agent/plugins/vppv2/aclplugin/vppcalls"
+	"github.com/ligato/vpp-agent/plugins/vppv2/ifplugin"
+	"github.com/ligato/vpp-agent/plugins/vppv2/model/acl"
+	"github.com/ligato/vpp-agent/plugins/vppv2/model/interfaces"
+	"github.com/pkg/errors"
+)
+
+const (
+	AclDescriptorName = "vpp-acl"
+)
+
+type AclDescriptor struct {
+	// dependencies
+	log        logging.Logger
+	aclHandler vppcalls.ACLVppAPI
+
+	// runtime
+	ifPlugin ifplugin.API
+	aclIndex aclidx.AclMetadataIndex
+}
+
+func NewAclDescriptor(aclHandler vppcalls.ACLVppAPI, ifPlugin ifplugin.API) *AclDescriptor {
+	return &AclDescriptor{
+		log:        logrus.NewLogger("acl-descriptor"),
+		ifPlugin:   ifPlugin,
+		aclHandler: aclHandler,
+	}
+}
+
+// GetDescriptor returns descriptor suitable for registration (via adapter) with
+// the KVScheduler.
+func (d *AclDescriptor) GetDescriptor() *adapter.AclDescriptor {
+	return &adapter.AclDescriptor{
+		Name: AclDescriptorName,
+		KeySelector: func(key string) bool {
+			return strings.HasPrefix(key, acl.Prefix)
+		},
+		ValueTypeName: proto.MessageName(&acl.Acl{}),
+		KeyLabel: func(key string) string {
+			name, _ := interfaces.ParseNameFromKey(key)
+			return name
+		},
+		ValueComparator: d.EquivalentACLs,
+		NBKeyPrefix:     acl.Prefix,
+		WithMetadata:    true,
+		MetadataMapFactory: func() idxmap.NamedMappingRW {
+			return aclidx.NewAclIndex(logrus.DefaultLogger(), "vpp-acl-index")
+		},
+		Add:                d.Add,
+		Delete:             d.Delete,
+		Modify:             d.Modify,
+		ModifyWithRecreate: d.ModifyWithRecreate,
+		IsRetriableFailure: d.IsRetriableFailure,
+		Dependencies:       d.Dependencies,
+		DerivedValues:      d.DerivedValues,
+		Dump:               d.Dump,
+		DumpDependencies:   []string{ /* ? */ },
```

Comment:
You will need the VPP-ifplugin to be dumped first, so that the ifaceidx is up-to-date for ACL-dump (you will need to translate indexes of ACL ingress/egress interfaces into logical names).